### ISSUE TITLE
Event [instance]_start never gets fired

### ISSUE THREAD

**Issue by gatonero** [jeroenterheerdt/HAsmartirrigation#181]
### What happened?

An automation should start when [instance]_start is fired. This never happens.

### How to reproduce

Use [instance]_start as a trigger for an automation.

### Relevant log output

```shell
None
```

### Which version are you running?

v2023.8.0-beta21

### Diagnostics file

- [X] I have attached a diagnostics file

### Additional information

[config_entry-smart_irrigation-6f22650b100cefed9d7411c48a4ba2b7.json.txt](https://github.com/jeroenterheerdt/HAsmartirrigation/files/12415760/config_entry-smart_irrigation-6f22650b100cefed9d7411c48a4ba2b7.json.txt)

**Comment by jeroenterheerdt** [comment]
please check the logs. It should say: '...start is scheduled for X time'. Check the name of the event and the time it fires. It does fire on my end.

**Comment by gatonero** [comment]
Here are my only scheduled events for smart irrigation. But I restarted my system this morning some hours after sunrise, so I'll check tomorrow morning again.

```
2023-08-23 16:27:48.270 INFO (MainThread) [custom_components.smart_irrigation] Scheduled auto update first time update for 4:00
2023-08-23 16:27:49.218 INFO (MainThread) [custom_components.smart_irrigation] Scheduled auto calculate update for 23:00
2023-08-23 16:27:49.219 INFO (MainThread) [custom_components.smart_irrigation] Scheduled auto update first time update for 4:00
```

**Comment by jeroenterheerdt** [comment]
so this doesn't say anything about when the start event will fire. that should be in the logs as well!

**Comment by jeroenterheerdt** [comment]
Did it fire?

**Comment by gatonero** [comment]
No, it didn't. Same log entries as before.

```
2023-08-24 16:34:20.841 INFO (MainThread) [custom_components.smart_irrigation] Scheduled auto update first time update for 4:00
2023-08-24 16:34:20.843 INFO (MainThread) [custom_components.smart_irrigation] Scheduled auto calculate update for 23:00
```

**Comment by jeroenterheerdt** [comment]
Weird. There should also be something in the log about start event will fire at...

**Comment by gatonero** [comment]
So I found something for the first time. 

`2023-08-24 16:34:20.658 INFO (MainThread) [custom_components.smart_irrigation] Start irrigation event smart_irrigation_start_irrigation_all_zones will fire at 2023-08-25 04:24:09.496869+00:00`

So I will close this issue.

**Comment by jeroenterheerdt** [comment]
Well, don't be too hasty.... Did it actually fire?

**Comment by gatonero** [comment]
Not until today. 

My three automations look like this.

```

alias: Bewässerung Wohnzimmer
description: ""
trigger:
  - platform: event
    event_type: smart_irrigation_start
  - platform: sun
    event: sunrise
    offset: "30"
condition:
  - condition: numeric_state
    entity_id: sensor.bewasserung_wohnzimmer
    above: 0
action:
  - service: switch.turn_on
    data: {}
    target:
      entity_id: switch.bewasserung_wohnzimmer
  - delay:
      seconds: "{{states(\"sensor.bewasserung_wohnzimmer\")}}"
  - service: switch.turn_off
    data: {}
    target:
      entity_id:
        - switch.bewasserung_wohnzimmer
  - service: smart_irrigation.reset_bucket
    data: {}
    target:
      entity_id:
        - sensor.bewasserung_wohnzimmer
mode: single

```
 The sun trigger is to run the irrigation for sure  and should fire after the start event. Until today my automations caught  only the time trigger.  

 Hopefully it catches the event trigger tomorrow morning.

**Comment by gatonero** [comment]
Just now I wondered that formatting of  

`seconds: "{{states(\"sensor.bewasserung_wohnzimmer\")}}"`

It is because seconds expect float. Maybe the format for that sensor could be adapted?

**Comment by jeroenterheerdt** [comment]
> Just now I wondered that formatting of
> 
> `seconds: "{{states(\"sensor.bewasserung_wohnzimmer\")}}"`
> 
> It is because seconds expect float. Maybe the format for that sensor could be adapted?

Not sure what you mean? Seconds is already a float but you can just cast it to a float by adding |float()

**Comment by jeroenterheerdt** [comment]
> Not until today.
> 
> My three automations look like this.
> 
> ```
> 
> alias: Bewässerung Wohnzimmer
> description: ""
> trigger:
>   - platform: event
>     event_type: smart_irrigation_start
>   - platform: sun
>     event: sunrise
>     offset: "30"
> condition:
>   - condition: numeric_state
>     entity_id: sensor.bewasserung_wohnzimmer
>     above: 0
> action:
>   - service: switch.turn_on
>     data: {}
>     target:
>       entity_id: switch.bewasserung_wohnzimmer
> delay:
>   seconds: "{{states(\"sensor.bewasserung_wohnzimmer\")}}"
>     data: {}
>     target:
>       entity_id:
>         - switch.bewasserung_wohnzimmer
>   - service: smart_irrigation.reset_bucket
>     data: {}
>     target:
>       entity_id:
>         - sensor.bewasserung_wohnzimmer
> mode: single
> ```
> 
> The sun trigger is to run the irrigation for sure and should fire after the start event. Until today my automations caught only the time trigger.
> 
> Hopefully it catches the event trigger tomorrow morning.

OK, let me know if the event doesn't trigger for you

**Comment by gatonero** [comment]
The irrigations started0 triggered  by sun. No start event.

**Comment by gatonero** [comment]
For the duration: I would like to keep things simple. I guess a lot of users use the UI to set up an automation. Because you can't enter a template there you have to edit it in YAML mode.
The format   
```delay:
  seconds: "{{states(\"sensor.bewasserung_wohnzimmer\")}}"
```
is very tricky for an user with little experience because of the backslashes.

**Comment by jeroenterheerdt** [comment]
> For the duration: I would like to keep things simple. I guess a lot of users use the UI to set up an automation. Because you can't enter a template there you have to edit it in YAML mode. The format
> 
> ```
>   seconds: "{{states(\"sensor.bewasserung_wohnzimmer\")}}"
> ```
> 
> is very tricky for an user with little experience because of the backslashes.

I have never used the backslashes myself and they aren't necessary

**Comment by gatonero** [comment]
Ok, I see no \ and no " works. 

```
delay:
  seconds: "{{states(sensor.bewasserung_wohnzimmer)}}"

```
I asked about the documentation.

**Comment by nelbs** [comment]
I have exactly the same problem..... 

When I changed the time the logs says
```
Logger: homeassistant.config_entries
Source: helpers/entity_component.py:187
First occurred: August 25, 2023 at 18:03:05 (57 occurrences)
Last logged: 00:33:38


Error setting up entry Irrigatie border for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 388, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 132, in async_setup_entry
    return await component.async_setup_entry(entry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 187, in async_setup_entry
    raise ValueError("Config entry has already been setup!")
ValueError: Config entry has already been setup!
```

**Comment by jeroenterheerdt** [comment]
> I have exactly the same problem.....
> 
> When I changed the time the logs says
> 
> ```
> Logger: homeassistant.config_entries
> Source: helpers/entity_component.py:187
> First occurred: August 25, 2023 at 18:03:05 (57 occurrences)
> Last logged: 00:33:38
> 
> 
> Error setting up entry Irrigatie border for sensor
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/config_entries.py", line 388, in async_setup
>     result = await component.async_setup_entry(hass, self)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 132, in async_setup_entry
>     return await component.async_setup_entry(entry)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 187, in async_setup_entry
>     raise ValueError("Config entry has already been setup!")
> ValueError: Config entry has already been setup!
> ```

This looks like you're running the v1. Please confirm your version number as this thread is about the version 2 (in beta)

**Comment by nelbs** [comment]
I am running version v0.0.81

**Comment by jeroenterheerdt** [comment]
> I am running version v0.0.81

The you don't have the exact same issue as this is about the beta. Also, I am no longer working v1 it will be replaced by v2 very soon.

**Comment by jeroenterheerdt** [comment]
@gatonero please try beta27 and see if the event fires now. I am hopeful it fires this time.

**Comment by gatonero** [comment]
Hopefully. Yesterday we have had a lot of rain. The runtime will be zero.

**Comment by jeroenterheerdt** [comment]
So that's an interesting one you might actually see the event firing at sunrise then... Oops 😂

**Comment by gatonero** [comment]
It's ten minutes passed sunrise. I see the event smart_irrigation_start under developer tools -> events with five active listeners. I also made an automation:

```
alias: Start Event smart irrigation
description: ""
trigger:
  - platform: event
    event_type: smart_irrigation_start
condition: []
action:
  - service: persistent_notification.create
    data:
      message: "Start Event fired. "
mode: single

```

but nothing to see.  I'm on v2023.8.0-beta27 :-(

**Comment by omenek159** [comment]
What you see in dev tools is only listeners, it doesn't tell you if something will trigger event or not. If you're on beta then event is called [instance]_start_irrigation_all_zones.

**Comment by gatonero** [comment]
The only event I see in events is [instance]_start.

![grafik](https://github.com/jeroenterheerdt/HAsmartirrigation/assets/8248255/5520839d-c12d-46a0-9e26-77a9f76adc64)

**Comment by jeroenterheerdt** [comment]
So wait, are you saying @omenek159 that the event fired for you even before beta 27??

**Comment by jeroenterheerdt** [comment]
> Thanks! Now I have seen it in the source code.
> 
> At the moment I'm writing some blueprints and I wonder how to integrate that trigger, means how to get [instance]?

Ask people for it?

**Comment by omenek159** [comment]
I have irrigation automation done in NR that uses [instance]_start_irrigation_all_zones and it worked at beta24. I don't know if it was working earlier because I was using time based trigger instead of this event. Even in v1 I did updates, calculating and activating irrigation at scheduled time.

**Comment by jeroenterheerdt** [comment]
Can you confirm it works still in this beta?

**Comment by omenek159** [comment]
Yes, event is firing in beta27.

![irr_nr](https://github.com/jeroenterheerdt/HAsmartirrigation/assets/10829566/1df16ec3-1ff5-4ac3-a136-4294efb848fe)
![irr_log](https://github.com/jeroenterheerdt/HAsmartirrigation/assets/10829566/3325901b-008b-45a9-bf4c-670cf0ef9cb0)

**Comment by jeroenterheerdt** [comment]
> Yes, event is firing in beta27.
> 
> ![irr_nr](https://user-images.githubusercontent.com/10829566/264929331-1df16ec3-1ff5-4ac3-a136-4294efb848fe.jpg) ![irr_log](https://user-images.githubusercontent.com/10829566/264929332-3325901b-008b-45a9-bf4c-670cf0ef9cb0.jpg)
> 
> But weird thing is when logging bucket reset, instead of showing zone name it shows each letter in different row.

That is weird indeed! @gatonero since you assigned this yourself are you going to take a look at this?

**Comment by gatonero** [comment]
Sorry, I didn't want that. I tried a new extension in vscode and wasn't clear about a function.

**Comment by jeroenterheerdt** [comment]
> Sorry, I didn't want that. I tried a new extension in vscode and wasn't clear about a function.

all good, just asking :)

**Comment by jeroenterheerdt** [comment]
> 

I am not seeing this on my end. Can you try trigger that service from HA itself instead of through NR? I am wondering if NR is doing something weird here.

**Comment by omenek159** [comment]
Hmm... Calling service from HA works ok, but there is something wrong with NR. Not only it logs like I showed you but also it won't reset buckets. Let me dig it more, but looks like there is nothing wrong with the component itself.

**Comment by jeroenterheerdt** [comment]
Closing this as event seems to fire. Please open another issue in case you are seeing something else.
